apiserver/facades/agent/provisioner/provisioninginfo.go

```diff
@@ -486,8 +487,21 @@ func (api *ProvisionerAPI) subnetsAndZonesForSpace(machineID string, spaceName s
 
 		zones := subnet.AvailabilityZones()
 		if len(zones) == 0 {
```

Comment:
Here too. Might it not be better to indicate whether AZs confine subnet availability via a method on `NetworkingEnviron`, which would allow us to design out this special casing?